Fetch Repositories

KBandipo opened the pull request:
This pull request makes the following changes 

- Fixes (https://discord.com/channels/1148391778492354741/1253584233155395657/1253594458172297270)

Add functionality to fetch the list of all public repositories on GitHub, alongside their repository links using Get request and display them on the UI upon page load.

KBandipo commented:
Hello @collabo-community/review-web-app , my PR is ready for review. Thank you

Ifycode commented on .env.example:
```diff
@@ -5,4 +5,7 @@ PORT=
+GITHUB_API_URL=
+
+# GITHUB REPOSITORIES API URL
+GITHUB_REPOS_API_URL=https://api.github.com/user/repos
```
Hmm... Let's see if we can reuse variables inside of the .env file @KBandipo
- See stackoverflow comment: https://stackoverflow.com/a/67615097
- Additional resource just incase: https://stackoverflow.com/a/48794954

KBandipo commented:
My pull request is ready for review. I have reused the base url variable inside of the .env file. This is the link to the task issue https://discord.com/channels/1148391778492354741/1253584233155395657/1253594458172297270
Thank you

Ifycode commented on .env.example:
```diff
@@ -4,5 +4,12 @@ PORT=
+GITHUB_API_URL=${BASE_URL}/repos/your-own-github-account-user-name-will-be-here-instead/the-name-of-the-repo-you-want-to-interact-with
+
+# Github Repositories API URL 
+GITHUB_REPOS_API_URL=${BASE_URL}/user/repos
```
You can just remove this one from here entirely, then you can use it in the code directly like this:

`${process.env.GITHUB_API_BASE_URL}/user/repos`

Ifycode reviewed (changes requested):
Nice Work @KBandipo 
Let's just clean it up a bit. See comments below.

Ifycode reviewed (changes requested):
@KBandipo always link your issue to your PR. Add your issue link approriately.

Ifycode commented:
@KBandipo I have the same question to ask you. the one that I asked Daniel. 
The link to your issue is in the wrong place. How do you expect reviewers/users to find it where you currently added it?

Ifycode commented on package-lock.json:
```diff
@@ -10,10 +10,10 @@
       "license": "ISC",
       "dependencies": {
```
package-lock.json should not be showing in your file changes @KBandipo

KBandipo replied:
Alright, @Ifycode  please how can I restore it?

KBandipo commented on .env.example:
```diff
@@ -4,5 +4,12 @@ PORT=
-GITHUB_API_URL=
\ No newline at end of file
+# Github Base URL
+BASE_URL=https://api.github.com
```
Okayyyy

KBandipo commented on .env.example:
```diff
@@ -4,5 +4,12 @@ PORT=
+BASE_URL=https://api.github.com
+
+# Supply your own github account user name and the name of the repo you want to interact with
+GITHUB_API_URL=${BASE_URL}/repos/your-own-github-account-user-name-will-be-here-instead/the-name-of-the-repo-you-want-to-interact-with
```
Copy boss

KBandipo commented:
Howdy boss, I have restored the package-lock.json as you requested @collabo-community/review-collabocate . I'm still working on the other things you corrected too. Thank you

KBandipo commented:
Hello @collabo-community/review-collabocate , my pull request is ready for review. These are my changes as you have requested; I replaced `BASE_URL` with `GITHUB_API_BASE_URL` and `GITHUB_API_URL` to `REPO_URL` in `.env.example`;  and Refactored the usage of Github Repositories API URL to `process.env.GITHUB_API_BASE_URL` directly in the server.js

tylerjusfly commented:
It says that you have a conflict, can you resolve that @KBandipo ?

KBandipo commented:
@tylerjusfly Thank you boss, I may not be able to resolve the conflict myself since I am not the only one working on the old server. @Ifycode is managing the issues together

Ifycode reviewed (changes requested):
@KBandipo you are yet to link your issue to the PR properly, I'm still waiting for this before I can touch/review your PR.

#

> About conflicts, @tylerjusfly I'll be the one to resolve the conflicts (it's caused by the restructuring I'm doing based all our 4 different PRs), so @KBandipo should not worry about the conflicts in this case

KBandipo commented:
Thank you @Ifycode , I have linked the issue to the pull request now. Please let me know if there are any additional steps I need to take

Ifycode commented:
@all-contributors add @KBandipo for code
